bundles/org.openhab.binding.speedtest/src/main/java/org/openhab/binding/speedtest/internal/SpeedtestHandler.java

```diff
@@ -0,0 +1,536 @@
+                case LINUX:
+                case MAC:
+                case SOLARIS:
+                    speedTestCommand = "/usr/bin/speedtest";
+                    break;
+                default:
+                    speedTestCommand = "";
+            }
+        }
+
+        updateStatus(ThingStatus.UNKNOWN);
+
+        if (!checkConfig(speedTestCommand)) { // check the config
+            return;
+        }
+        if (!getSpeedTestVersion()) {
+            return;
+        }
+        getServerList();
+        updateStatus(ThingStatus.ONLINE);
+        isRunning = true;
+        onUpdate(); // Setup the scheduler
+    }
+
+    /**
+     * This is called to start the refresh job and also to reset that refresh job when a config change is done.
+     */
+    private void onUpdate() {
+        logger.debug("Polling Interval Set: {} ", pollingInterval);
+        if (pollingInterval > 0) {
+            if (pollingJob == null || pollingJob.isCancelled()) {
+                pollingJob = scheduler.scheduleWithFixedDelay(pollingRunnable, 0, pollingInterval, TimeUnit.MINUTES);
+            }
+        }
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Disposing Speedtest Handler Thing");
+        isRunning = false;
+        if (pollingJob != null && !pollingJob.isCancelled()) {
+            pollingJob.cancel(true);
+            pollingJob = null;
+        }
+    }
+
+    /**
+     * Called when this thing gets it's configuration changed.
+     */
+    @Override
+    public void thingUpdated(Thing thing) {
+        dispose();
+        this.thing = thing;
+        initialize();
+    }
+
+    /**
+     * Polling event used to get speed data from speedtest
+     */
+    private Runnable pollingRunnable = () -> {
+        try {
+            getSpeed();
+        } catch (Exception e) {
+            logger.warn("An exception occurred while running Speedtest: '{}'", e.getMessage());
+            updateStatus(ThingStatus.OFFLINE);
+        }
+    };
+
+    /*
+     * Gets the version information from speedtest, this is really for debug in the event they change things
+     */
+    private boolean getSpeedTestVersion() {
+        String versionString = doExecuteRequest(" -V", String.class);
+        if ((versionString != null) && !versionString.isEmpty()) {
+            int newLI = versionString.indexOf("\n");
+            String versionLine = versionString.substring(0, newLI);
+            if (versionString.indexOf("Speedtest by Ookla") > -1) {
+                logger.debug("Speedtest Version: {}", versionLine);
+                return true;
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "Unsupported type of Speedtest recognized, Speedtest CLI tool from Ookla is REQUIRED (https://www.speedtest.net/). Please check installation/configuration.");
+                return false;
+            }
+        }
+        return false;
+    }
+
+    /*
```

Comment:
Also check the others.

```suggestion
    /**
```

Reply:
changed all